packages/wonder-blocks-form/src/components/labeled-field.js

```diff
@@ -0,0 +1,294 @@
+     * upon error.
+     * Note: The string will not be used if a `validate` prop is passed in.
+     *
+     * Example message: i18n._("A password is required to log in.")
+     *
+     * Boolean:
+     * True/false indicating whether this field is required. Please do not pass
+     * in `true` if possible - pass in the error string instead.
+     * If `true` is passed, and a `validate` prop is not passed, that means
+     * there is no corresponding message and the default untranlsated message
+     * will be used.
+     */
+    required?: boolean | string,
+
+    /**
+     * Identifies the element or elements that describes this form field.
+     */
+    ariaDescribedby?: string | Array<string>,
+
+    /**
+     * Custom styles for the container.
+     */
+    style?: StyleType,
+
+    /**
+     * Optional test ID for e2e testing.
+     */
+    testId?: string,
+
+    /**
+     * Whether to validate the field value on change.
+     */
+    instantValidation?: boolean,
+
+    /**
+     * Provide a validation for the field element.
+     * Returns a string error message, a promise with a string error message or
+     * null | void for a valid result.
+     */
+    validate?:
+        | ((value: string) => MaybeAsyncString)
+        | ((Array<string>) => MaybeAsyncString),
+
+    /**
+     * Called right after the field is validated. Note that validation is called
+     * on change.
+     */
+    onValidate?: (errorMessage: ?string) => mixed,
+
+    layout?: "stacked" | "horizontal" | "inline",
+|};
+
+/**
+ * A LabeledField is an element used to accept a single line of text from the
+ * user paired with a label, description, and error field elements.
+ *
+ * ### Usage
+ *
+ * ```jsx
+ * import {LabeledField} from "@khanacademy/wonder-blocks-form";
+ *
+ * const [value, setValue] = React.useState("");
+ *
+ * <LabeledField
+ *     label="Label"
+ *     description="Hello, this is the description for this field"
+ *     field={<SingleSelect onChange={setValue} value={value} />}
+ *     required={true}
+ *     validate={(value => {
+ *         if (!value) {
+ *              return "Please select an option";
+ *         }
+ *      })}
+ * />
+ * ```
+ */
+const LabeledField: React.AbstractComponent<Props, HTMLElement> =
+    React.forwardRef<Props, HTMLElement>(function LabeledField(
+        props: Props,
+        ref,
+    ) {
+        const {
+            ariaDescribedby,
+            description,
+            field,
+            id,
+            label,
+            instantValidation = true,
+            required,
+            style,
+            testId,
+            onValidate,
+            validate,
+        } = props;
+
+        const ids = useUniqueIdWithMock("labeled-field");
+        const uniqueId = id ?? ids.get("labeled-field-id");
+
+        const [error, setError] = React.useState(null);
+        const [active, setActive] = React.useState(false);
+
+        useOnMountEffect(() => {
```

Comment:
@jandrade this part is where it feels hacky to me. how are we enforcing what the name of the value field is on the `field` component? we already need a new one for `checked` for radio boxes